### ISSUE TITLE
Example data type mismatch

### ISSUE THREAD

**Issue by linonetwo** [iannbing/react-simple-tree-menu#19]
```js
const treeData = [
    {
      key: 'first-level-node-1',
      label: 'Node 1 at the first level',
      nodes: [
        {
          key: 'second-level-node-1',
          label: 'Node 1 at the second level',
          nodes: [
            {
              key: 'third-level-node-1',
              label: 'Last node of the branch',
              nodes: [] // you can remove the nodes property or leave it as an empty array
            },
          ],
        },
      ],
    },
    {
      key: 'first-level-node-2',
      label: 'Node 2 at the first level',
    },
  ];
  
  return (
    <>
        <TreeMenu data={treeData} />
```

```
(JSX attribute) data?: {
    [name: string]: TreeNode;
} | undefined
Type '({ key: string; label: string; nodes: { key: string; label: string; nodes: { key: string; label: string; nodes: never[]; }[]; }[]; } | { key: string; label: string; nodes?: undefined; })[]' is not assignable to type '{ [name: string]: TreeNode; }'.
  Index signature is missing in type '({ key: string; label: string; nodes: { key: string; label: string; nodes: { key: string; label: string; nodes: never[]; }[]; }[]; } | { key: string; label: string; nodes?: undefined; })[]'.ts(2322)
index.d.ts(5, 5): The expected type comes from property 'data' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<TreeMenu> & 
Pick<Readonly<TreeMenuProps> & Readonly<{ children?: ReactNode; }>, never> & Partial<Pick<Readonly<TreeMenuProps> & Readonly<...>, "data" | ... 4 more ... | "debounceTime">> & Partial<...>'
```

**Comment by linonetwo** [comment]
This don't works either:

```js
  const treeData = [
    {
      key: 'first-level-node-1',
      label: '铆钉总数 150 个  检测到铆钉 150 个',
      index: 0,
      nodes: [
        {
          key: 'first-level-node-1',
          label: '有 12 个铆钉需要复检',
          index: 0,
          nodes: [
            {
              key: 'second-level-node-1',
              label: '5mm 铆钉 #33 大小不匹配',
              index: 0,
            },
            {
              key: 'second-level-node-2',
              label: '8mm 铆钉 #43 漏打',
              index: 1,
            },
          ],
        },
      ],
    },
  ];
```

But if `<TreeMenu data={treeData as any} />` then it works.

**Comment by iannbing** [comment]
Thanks for reporting. It has been fixed in this PR #20  and released as `1.0.3`.
